### ISSUE TITLE
What is host and user

### ISSUE THREAD

**Issue by ca1p3** [johnnybegood/ha-ksenia-lares#29]
Hi this isnt any issue but i dont understand what the host is and if the username and password is the things i type when i login to ksenia secureweb

**Comment by johnnybegood** [comment]
The host is the IP or domain name of the system. Or the part between the :// and the first / in the URL of the web client.

It is the same user name and password

**Comment by ca1p3** [comment]
> The host is the IP or domain name of the system. Or the part between the :// and the first / in the URL of the web client.
> 
> It is the same user name and password

when i enter the ip adress and let the port be 4202 and using my email and password as i use to sign into ksenia secureweb i get Invalid authentication. is there something i need to do in secureweb first ?

**Comment by johnnybegood** [comment]
> > The host is the IP or domain name of the system. Or the part between the :// and the first / in the URL of the web client.
> 
> > 
> 
> > It is the same user name and password
> 
> 
> 
> when i enter the ip adress and let the port be 4202 and using my email and password as i use to sign into ksenia secureweb i get Invalid authentication. is there something i need to do in secureweb first ?

I am not a user of the secureweb, so can't say if this is the same.  If you connect to http://LOCAL-IP:port, it is the user name and password used there.

This a full local integration, it doesn't use any cloud connector.

**Comment by ca1p3** [comment]
> > > The host is the IP or domain name of the system. Or the part between the :// and the first / in the URL of the web client.
> > 
> > 
> > > 
> > 
> > 
> > > It is the same user name and password
> > 
> > 
> > when i enter the ip adress and let the port be 4202 and using my email and password as i use to sign into ksenia secureweb i get Invalid authentication. is there something i need to do in secureweb first ?
> 
> I am not a user of the secureweb, so can't say if this is the same. If you connect to http://LOCAL-IP:port, it is the user name and password used there.
> 
> This a full local integration, it doesn't use any cloud connector.

when i enter the ip adress to ksenia and use the port 4042 nothing happens, i cant reach the site.

**Comment by johnnybegood** [comment]
Most likely the local webserver is not enabled by the installer. Or it is setup on another port.

The installer should be able to help you with this info, if you can login to the local webserver then the integration should work.

**Comment by ca1p3** [comment]
> Most likely the local webserver is not enabled by the installer. Or it is setup on another port.
> 
> The installer should be able to help you with this info, if you can login to the local webserver then the integration should work.

I just realised that im using lares 4.0 does this work with my card

**Comment by johnnybegood** [comment]
It is made for the previous generation the Lares 48 IP and variants. The 4.0 is not tested and likely not compatible.

**Comment by johnnybegood** [comment]
It looks like the 4.0 has a web interface under port 443, so https://LOCAL-IP should give something.

**Comment by ca1p3** [comment]
> It looks like the 4.0 has a web interface under port 443, so https://LOCAL-IP should give something.

yes it does the problem is that i come to my loginpage but there is only a pin code i enter to accsess my interface. a 6 digit number.

**Comment by johnnybegood** [comment]
Do you see anything when opening `https://LOCAL-IP/xml/info/generalInfo.xml` ?

**Comment by ca1p3** [comment]
> Do you see anything when opening `https://LOCAL-IP/xml/info/generalInfo.xml` ?

I Will try and reply when i come Home

**Comment by ca1p3** [comment]
> Do you see anything when opening `https://LOCAL-IP/xml/info/generalInfo.xml` ?

I get this when I use the adress with the ip adress and :433 
panel WebServer 404 Not Found

The requested document was not found on this server.

**Comment by samuelemarazzita** [comment]
Same issue here, there is no username and password, only a pin, also as an installer.

**Comment by johnnybegood** [comment]
@samuelemarazzita / @ca1p3 if there is no response on `https://LOCAL-IP/xml/info/generalInfo.xml` than the component can't talk to the alarm. I don't have Lares 4.0 to test with.

**Comment by samuelemarazzita** [comment]
This is shown when I visit the URL `https://LOCAL-IP/xml/info/generalInfo.xml`

<img width="2056" alt="image" src="https://github.com/user-attachments/assets/a1cdd478-8300-45b8-adea-fa58a747d1cb">

**Comment by ca1p3** [comment]
Same for me

**Comment by johnnybegood** [comment]
Sorry,

The 4.0 version works completely different and is not support. I would like to add support but I have no 4.0 available to see how it works.

**Comment by MontaniDaniele** [comment]
> Sorry,
> 
> The 4.0 version works completely different and is not support. I would like to add support but I have no 4.0 available to see how it works.

Hi, I also just installed a new alarm with lares 4.0, do you think there could be some development in the near future to be able to integrate it into this component or not?

**Comment by ca1p3** [comment]
> Sorry,
> 
> The 4.0 version works completely different and is not support. I would like to add support but I have no 4.0 available to see how it works.

have you got your hands on a 4.0 yet ?

**Comment by johnnybegood** [comment]
No, and as it a physical alarm system that isn't very likely. More likely scenario is someone with a 4.0 that can give me some insight on how it works.

Although from what I see online, I fear that the only things in common is the name.

**Comment by emanuelegreco29** [comment]
For anyone wondering, there is a custom integration made by Sdomotica. However, it only works on RaspberryPi, not on VMs or native HomeAssistant OS on bare metal ARM64. I hope someone can find an alternative, as that integration has a price of 99$.

**Comment by dextorer** [comment]
I've started to look into this. The new Lares 4.0 is quite different from its predecessors (uses websockets for communication), but I was able to get a proof-of-concept working, i.e. logging in with my PIN code and get a response from the server with my user data.

There are a few ways to go about this:
1. I could work with @johnnybegood and add support together
2. I could use this integration as a base, then change the communication protocol to adapt it to the new 4.0 system
3. I could write a new integration from scratch

@johnnybegood How would you feel about the above? My goal would of course be to have it for myself first, even better if I can share it with the wider community 👍

**Comment by johnnybegood** [comment]
@dextorer If you have a base version working, we could for a way together to make it compatible with both. I should also have look then at splitting of Lares python library.

**Comment by emanuelegreco29** [comment]
If you decide to join forces and a new repo is created, could you please share it here. I’m starting to learn my way into HA integrations, I have some python knowledge and I’m willing to help if needed, since I also have a Keshia 4.0 Lares setup in my home.

**Comment by johnnybegood** [comment]
Lets join forces, I'll have a look at starting to split it off. In the mean time you an experiment with changing the code in `base.py` https://github.com/johnnybegood/ha-ksenia-lares/blob/49d59ce432913b30ca9b642508d6f45cfeaf7f99/custom_components/ksenia_lares/base.py#L17 

All logic for connecting is in this class.

**Comment by johnnybegood** [comment]
Moving to https://github.com/johnnybegood/ksenia-lares/issues/2 for tracking.
